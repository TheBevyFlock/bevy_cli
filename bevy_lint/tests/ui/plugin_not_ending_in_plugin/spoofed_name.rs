//! A test that ensures a plugin whose name is "spoofed" with `use T as F` does not sneak past the
//! lint.

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::plugin_not_ending_in_plugin)]

use bevy::prelude::*;

mod bar {
    pub mod baz {
        pub struct Foo;
        //~^ HELP: rename the plugin
    }
}

// We try to be sneaky, but it doesn't work.
use self::bar::baz::Foo as FooPlugin;

//~v ERROR: implemented `Plugin` for a structure whose name does not end in "Plugin"
impl Plugin for FooPlugin {
    fn build(&self, _app: &mut App) {}
}

fn main() {
    App::new().add_plugins(FooPlugin);
}
