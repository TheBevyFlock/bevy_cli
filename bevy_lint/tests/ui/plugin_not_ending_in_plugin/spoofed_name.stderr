error: implemented `Plugin` for a structure whose name does not end in "Plugin"
  --> tests/ui/plugin_not_ending_in_plugin/spoofed_name.rs:21:1
   |
12 |           pub struct Foo;
   |                      --- help: rename the plugin: `FooPlugin`
...
21 | / impl Plugin for FooPlugin {
22 | |     fn build(&self, _app: &mut App) {}
23 | | }
   | |_^
   |
note: the lint level is defined here
  --> tests/ui/plugin_not_ending_in_plugin/spoofed_name.rs:6:9
   |
6  | #![deny(bevy::plugin_not_ending_in_plugin)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

